conn.go

```diff
@@ -218,6 +255,18 @@ func (c *connect) compressBuffer(start int) error {
 }
 
 func (c *connect) sendData(block *proto.Block, name string) error {
+	if c.isClosed() {
+		err := errors.New("attempted sending on closed connection")
+		c.debugf("[send data] err: %v", err)
+		return err
+	}
+
+	if c.buffer == nil {
+		err := errors.New("attempted sending on nil buffer")
+		c.debugf("[send data] err: %v", err)
+		return err
+	}
```

Comment:
is it something you found when while debugging or only added for safety?

Reply:
yup